File: 1894-merge-strings-alternately/1894-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string mergedWord = "";
        for(int i=0; ; ){
            if(i<word1.length())    mergedWord += word1[i];
            if(i<word2.length())    mergedWord += word2[i];
            i++;
            if(i>=word1.length() && i>=word2.length())  break;
        }
        return mergedWord;
    }
};

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder s = new StringBuilder();
        int i=0, j=0;
        while(i<word1.length() && j<word2.length()){
            s.append(word1.charAt(i));
            s.append(word2.charAt(j));
            i++;
            j++;
        }
        if(i==word1.length()){
            while(j<word2.length()){
                s.append(word2.charAt(j));
                j++;
            }
        }
        if(j==word2.length()){
            while(i<word1.length()){
                s.append(word1.charAt(i));
                i++;
            }
        }
        return s.toString();
    }
}